packages/inventory-compliance/src/SystemRulesTable.js

```diff
@@ -400,14 +400,15 @@ class SystemRulesTable extends React.Component {
     filterBy = (attribute, rows, column) => {
         const filteredRows = [];
         rows.forEach((row, i) => {
-            if (row.hasOwnProperty('isOpen') && String(row.cells[column].original).toLowerCase().match(
+            if (this.isParent(row) && String(row.cells[column].original).toLowerCase().match(
                 String(attribute).toLowerCase()
             )) {
                 filteredRows.push(row);
-                if (!rows[i + 1].hasOwnProperty('isOpen')) {
-                    let child = rows[i + 1];
-                    child.parent = filteredRows.length - 1;
-                    filteredRows.push(child);
+                if (!this.isParent(rows[i + 1])) {
+                    filteredRows.push({
```

Comment:
This is the gist of the fix.

Apparently, `child.parent = filteredRows.length - 1` was not respected properly, so the parent ID would not be 0, 2, 4, etc... it would be the parent ID of the row in the unfiltered array (so it could be something stupid like 12, 18...). 

Actually even doing `.push { parent: filteredRows.length - 1, ...rows[i + 1] }`, the parent attribute was still set to the `rows[i + 1]` attribute. I'm not even sure why, so I decided to pass `cells` explicitly and ignore the parent